syntax = "proto3";

option cc_generic_services = true;

import "google/protobuf/empty.proto";
package apb.api.data;

option java_package = "jp.co.apb.server.api.proto.data";
option java_multiple_files = true;
option csharp_namespace = "Art.Framework.ApiNetwork.Grpc.Api.Data";

message MasterDataGetLatestVersionResponse {
  string latest_master_data_version = 1;
}

message UserDataGetRequest {
  repeated string table_name = 1;
}

message UserDataGetResponse {
  map<string, string> user_data_json = 1;
}

message UserDataGetNameResponse {
  repeated string table_name = 1;
}

message UserDataGetNameResponseV2 {
  repeated .apb.api.data.TableNameList table_name_list = 1;
}

message TableNameList {
  repeated string table_name = 1;
}

message DiffData {
  string update_records_json = 1;
  string delete_keys_json = 2;
}

service DataService {
  rpc GetLatestMasterDataVersion(.google.protobuf.Empty) returns (.apb.api.data.MasterDataGetLatestVersionResponse);
  rpc GetUserDataName(.google.protobuf.Empty) returns (.apb.api.data.UserDataGetNameResponse);
  rpc GetUserDataNameV2(.google.protobuf.Empty) returns (.apb.api.data.UserDataGetNameResponseV2);
  rpc GetUserData(.apb.api.data.UserDataGetRequest) returns (.apb.api.data.UserDataGetResponse);
}

